test/observer.spec.ts

```diff
@@ -147,7 +146,7 @@ describe('DataLayerObserver unit tests', () => {
     });
 
     expect(observer.handlers.length).to.eq(1);
-    // @ts-ignore TODO (van) how to typecheck this
+
```

Comment:
tada, `ts-ignore` statements can be removed

Reply:
I think want you really want is 

```
registerOperator(name: string, operator: InstanceType<typeof Operator>) {
```

but in order to do that, you'll need to make the `Operator` non-abstract. That will require a little more gymnastics.

Reply:
@patrick-fs Thank you.  Looking at it now.  I think the dilemma that exists is that it may be difficult to enforce the type safety for module developers AND have "generics" to support a DSL using DataLayerRule.  Might have to pick the one that has less consequences for code maintenance, and that at present looks like moving to this type of a re-design.  Fortunately @TrevorFSmith had suggested a `validate` function so even if we move to generic options that we don't know the shape of, we can always use validate to ensure the options are semantically correct.  I believe there might be additional convenience functions needed to help validate rules easily so it's not a gnarly bunch of if/else blocks.  @TrevorFSmith, do you want to approve the existing #23 in isolation and incorporate these changes in another PR or simply make them now in #23 ?

Reply:
I'd prefer to handle them separately so I'll go review #23 now.